### DISCUSSION TITLE
"Recircle" common phrases between courses

### DISCUSSION THREAD

**Discussion by Pottit** [kantord/LibreLingo#957] [Ideas]
Hi,

I am very interested in following the development of this project, and believe it's potential impact could be incredible. As the project matures, I believe the ease of contributing new courses will be decisive, which is the basis of the idea presented below.

### Humble disclaimer
I don't know exactly what the plans are for the course editor, or how far along the development process is. But I've been thinking of how I could contribute when it is ready, ideally in a way that does not interrupt an already busy schedule. I get that this will probably not be a priority for a long time still (if ever), but I would just like to throw it out there in case it's something that could be accommodated for in this early stage of development.

Of course, I fully understand that challenge right now is to build the software itself rather than to come up with crazy ideas. That being said:

### The idea
It might not be very original, but in either case: My idea is that when beginning to contribute a course, instead of coming up with original phrases, the user is given basic phrases to translate from existing courses. This would not only make it much easier to build the foundation for a new course, but it would go a long way to solve the problem of different ways of writing the same sentence. Furthermore, it could make course creation itself a learning experience. 

It would also make the creation of a new course intuitive: Rather than starting with nothing, LibreLingo could ask the contributor to translate phrases like "the man", "an apple" etc., gradually working through a set of basic phrases. I believe this would lower the bar for contributions significantly.

### Course building as learning
If I am to daydream, this function could allow users to translate from a language that they are *not* completely familiar with. Say that I'm a native Norwegian speaker who is more or less fluent in English (as I believe has to be a precondition) and who is currently learning French. Furthermore, basic phrases have already been translated to French by some kind contributor.

Rather than translating English phrases to Norwegian (which would not be very stimulating), the course editor could present me with the French language versions.

Course creation could therefore itself be game-like, except that I would have to validate the answers myself.

> Please translate to Norwegian
>
> 🔈 **l'homme**
> > *mannen*        ← this would be the text field
>
> `Show English` `Contribute translation` ← these would be buttons

After pressing the "contribute" button the translator should be presented with a window confirming the recorded data from English. This would allow the user to submit synonyms/alternative phrases, or to correct themselves. This is important, both to avoid human errors and problems stemming from double meanings.

> You have contributed the following:
>
> English: **the man**
> Norwegian: **mannen**
>
> `There's a mistake` `Correct - next translation`
> `Add alternative spelling or translation`

### Benefit of alternative translations
One benefit of this approach becomes clear with a more complex phrase, such as "The man rides a bicycle".  By recirculating this phrase across courses, the fact that it could also be written as "the man cycles" would not have to be repeated for every new course, and missing translation within this proposed "core" set of phrases would be fewer as a consequence. 

Of course, I'm not proposing to standardize the entire platform - German learners will want to learn about lederhosen, and Italian learners will need a hundred words for ham. Not to mention language-specific challenges related to grammar, which would require a greater element of supervision. However, it would be a good way to ensure a solid basis for community-made courses. Maybe it could even make a real difference for some near-extinct languages. :)

Last, keep up the great work! :turtle:

**Comment by kantord** [comment]
Interesting! A similar idea might be "simply" introducing a new way to learn multiple courses at once, which would come with this feature.

Another similar idea might be "automatically" importing sentences from other courses. This would be importing easy sentences from another course by giving them to advanced users. If multiple advanced users coincide in the same translations, it would be automatically added to the easy sentences.

@allcontributors please add @Pottit for ideas

**Comment by Pottit** [comment]
I love the idea of (semi) automatically expanding the courses. Another approach would be to add an option for users to register a mother tongue, and to now and then ask them to contribute by translating a sentence to their language. Say i just successfully translated something from French to English, LibreLingo might ask me for a Norwegian translation of the same sentence, knowing I'm a native speaker. It should of course be optional to fill in and probably should not be prompted too often (as it's a distraction), but it should be too much to ask for once in a while.

There should probably also be a way of verifying users/individual translations before these translations are implemented into courses, to avoid abuse and/or incompetence...

As for learning multiple languages at once, I guess our brains operate in different ways - for me it would be guaranteed to lead to substantial confusion. I'm struggling enough to keep French and Italian separated in my head already... :sweat_smile:
